### PR TITLE
Document command altering process

### PR DESCRIPTION
**Pull request by adamzimmermann**
This tripped me up when I was trying to implement this. I forget where I ended up finding the example that got me unblocked, but figured I would help others avoid my confusion.

### CODE REVIEW

File: sut/modules/unish/woot/drush.services.yml [hunk 1]
```diff
@@ -17,3 +17,7 @@ services:
     class: Drupal\woot\Commands\AnnotatedGreetCommand
     tags:
       -  { name: console.command }
+  woot.command_alterer:
```
Comment (adamzimmermann) [inline sut/modules/unish/woot/drush.services.yml:20]: I'm unsure if you want this change, but my thought process was that this example should be complete. 🤔

File: docs/commands.md [hunk 2]
```diff
@@ -64,9 +64,16 @@ Drush command info (annotations/attributes) can be altered from other modules. T
...
+1. The filename must have a name like `FooCommandInfoAlterer.php`. The prefix `Foo` can be whatever string you want. The file must end in `CommandInfoAlterer.php`.
 1. The class must implement the `\Consolidation\AnnotatedCommand\CommandInfoAltererInterface`.
 1. Implement the alteration logic in the `alterCommandInfo()` method.
+1. Create a tagged service definition in `drush.services.yml` with the `drush.command_info_alterer` tag for the alterer class.
+   ```yaml
+   foo.command_alterer:
+     class: \Drupal\<module-name>\Drush\CommandInfoAlterers\FooCommandInfoAlterer
+     tags:
+       - { name: drush.command_info_alterer }
+   ```
```
Comment (apotek) [inline docs/commands.md:76]: Crucial missing link here. Nice sleuthing and this is definitely going to help others!

**Review by apotek — APPROVED** [review]
Vital change for completeness and will save others lots of time.

File: docs/commands.md [hunk 3]
```diff
@@ -64,9 +64,16 @@ Drush command info (annotations/attributes) can be altered from other modules. T
 In the module that wants to alter a command info, add a class that:
 
 1. The generator class namespace, relative to base namespace, should be `Drupal\<module-name>\Drush\CommandInfoAlterers` and the class file should be located under the `src/Drush/CommandInfoAlterers` directory.
-1. The filename must have a name like FooCommandInfoAlterer.php. The prefix `Foo` can be whatever string you want. The file must end in `CommandInfoAlterer.php`.
+1. The filename must have a name like `FooCommandInfoAlterer.php`. The prefix `Foo` can be whatever string you want. The file must end in `CommandInfoAlterer.php`.
 1. The class must implement the `\Consolidation\AnnotatedCommand\CommandInfoAltererInterface`.
 1. Implement the alteration logic in the `alterCommandInfo()` method.
+1. Create a tagged service definition in `drush.services.yml` with the `drush.command_info_alterer` tag for the alterer class.
```
Comment (weitzman) [inline docs/commands.md:70]: This is true in Drush 11, but drush.services.yml is deprecated in Drush 12. I dont think this is needed there. You  could change the target for this PR to 11.x and then copy some of the edits to a new PR fo 12.x branch. We test this feature in https://github.com/drush-ops/drush/blob/3d0d136ed3ceeb41d3a25e1e155b2bd655095c16/tests/functional/CommandInfoAlterTest.php so I'm pretty sure it is working in both branches.

Comment (adamzimmermann) [reply docs/commands.md:70]: I was using Drush 11, so that would make sense. I'll double check this in Drush 12 and update the PR.

Thank you for the feedback 🙏🏼

**Comment by adamzimmermann** [comment]
Closing and opening a new PR against the 11.x branch.

File: docs/commands.md (hunk 3 above)
Comment (adamzimmermann) [reply docs/commands.md:70]: I followed the logic in the 12.x branch and it all checks out. Very cool. 👍🏼 

I realized that you were already on top of this in the 11.x documentation, and I was just looking in the wrong spot. 🙈 

- https://github.com/drush-ops/drush/blob/11.x/docs/commands.md?plain=1#L89

tl;dr we should keep this PR closed and there is nothing more to do here.
